modules/decline/src/Smithy4sCli.scala

```diff
@@ -53,6 +53,7 @@ class Smithy4sCli[Alg[_[_, _, _, _, _]], F[_]: MonadThrow](
   ): List[String] =
     HttpEndpoint
       .cast(endpoint)
+      .toOption
```

Comment:
minor: since we're throwing a few lines below, would it be worth it to throw here also rather than just swallow the error?

Reply:
yeah lemme do that

Reply:
actually, I don't think so...

If `HttpEndpoint.cast` fails, that means the endpoint is _not_ a HTTP one and this method should just return `Nil`. We're throwing iff `HttpEndpoint.cast` succeeds and the `Http` hint is missing, which would be a completely invalid state.

The purpose of `protocolSpecificHelp` is to generate a list of additional help strings based on... protocol-specific info 😅 
so if you're implementing a CLI for a service that _only_ exposes a CLI (and no HTTP interface), you want `Nil` here.

Reply:
I do the same in Playground and I've been thinking about it recently: https://github.com/kubukoz/smithy-playground/issues/139 - basically, if protocol-specific interpreters are allowed to prefix URLs, maybe we shouldn't trust the `Http` hint. Or `HttpEndpoint`, for that matter. Not sure what to think about this, but it's a separate problem from this PR I think.